src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -173,3 +175,97 @@ g.test('readwrite_upon_aspects')
       encoder.finish();
     }, !success);
   });
+
+g.test('binding_types_combination')
+  .params([
+    { type0: 'sampled-texture' as const, type1: 'sampled-texture' as const, _success: true },
+    {
+      type0: 'sampled-texture' as const,
+      type1: 'readonly-storage-texture' as const,
+      _success: true,
+    },
+    {
+      type0: 'sampled-texture' as const,
+      type1: 'writeonly-storage-texture' as const,
+      _success: false,
+    },
+    { type0: 'sampled-texture' as const, type1: 'render-target' as const, _success: false },
+    {
+      type0: 'readonly-storage-texture' as const,
+      type1: 'readonly-storage-texture' as const,
+      _success: true,
+    },
+    {
+      type0: 'readonly-storage-texture' as const,
+      type1: 'writeonly-storage-texture' as const,
+      _success: false,
+    },
+    {
+      type0: 'readonly-storage-texture' as const,
+      type1: 'render-target' as const,
+      _success: false,
+    },
+    {
+      type0: 'writeonly-storage-texture' as const,
+      type1: 'writeonly-storage-texture' as const,
+      _success: true,
+    },
+    {
+      type0: 'writeonly-storage-texture' as const,
+      type1: 'render-target' as const,
+      _success: false,
+    },
+  ])
+  .fn(async t => {
+    const { type0, type1, _success } = t.params;
+
+    const view = t
+      .createTexture({
+        usage:
+          GPUTextureUsage.SAMPLED | GPUTextureUsage.STORAGE | GPUTextureUsage.OUTPUT_ATTACHMENT,
+      })
+      .createView();
+
+    const bglEntries: GPUBindGroupLayoutEntry[] = [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.FRAGMENT,
+        type: type0,
+        // Currently setting two 'render-target' usages in the same render pass is impossible.
+        // We don't need to test that. So type0 is either 'sampled-texture' or storage usages.
+        // It can't be 'render-target'.
+        storageTextureFormat: type0 === 'sampled-texture' ? undefined : 'rgba8unorm',
```

Comment:
This is an example where setting information on the bind group layout entry would have been useful even if it isn't used by this `GPUBindingType`.

Reply:
What are you talking about, it seems that it is related to another issue? :)

Reply:
oops, sorry for the lack of context. Part of the discussion: https://github.com/gpuweb/gpuweb/issues/829, https://github.com/gpuweb/gpuweb/pull/949#issuecomment-664365066
I think there was more elsewhere, but I don't remember where (maybe in a meeting).

Reply:
Thank you, Kai. This is useful in CTS because it is easier for us to iterate over each binding type. But I suppose we may rarely use it in real application because we know exactly the particular binding type in most cases.